File: LeetCode/동적 프로그래밍/Pascal's Triangle.py
# https://leetcode.com/problems/pascals-triangle/description/
from typing import List

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle = []
        for row_num in range(numRows):
            row = [1 for _ in range(row_num + 1)]  # 현재 행의 모든 요소를 1로 초기화
            
            # 현재 행의 가운데 숫자 계산
            for j in range(1, len(row) - 1):
                row[j] = triangle[row_num - 1][j - 1] + triangle[row_num - 1][j] # 이전 행의 j-1번째 요소와 j번째 요소를 더함
            
            triangle.append(row)
        
        return triangle

File: LeetCode/동적 프로그래밍/Pascal's Triangle.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> row = new ArrayList<>();
        
        // rowIndex + 1 길이만큼 초기화
        for (int i = 0; i <= rowIndex; i++) {
            row.add(0);
        }
        row.set(0, 1); // 첫 번째 값은 항상 1

        for (int i = 1; i <= rowIndex; i++) {
            // 오른쪽 → 왼쪽 방향으로 업데이트
            for (int j = i; j > 0; j--) {
                row.set(j, row.get(j) + row.get(j - 1));
            }
        }
        
        return row;
    }
}
